### PR TITLE
simplify hashing

### PR DESCRIPTION
**Pull request by chungg**
we don't need math to build a hash

### CODE REVIEW

**Comment by chungg** [comment]
i should probably add, this does next to nothing with regards to performance.

File: gnocchi/carbonara.py [hunk 1]
```diff
@@ -453,11 +453,8 @@ def __iter__(self):
         return self
 
     def __hash__(self):
-        return hash(
-            str(datetime64_to_epoch(self.key))
-            +
-            str(self.sampling / ONE_SECOND)
-        )
+        return hash(str(self.key.astype('datetime64[ns]')) +
```
Comment (jd) [inline gnocchi/carbonara.py]: So it seems that it's better to use `view` than `astype`: `view` does not return a copy, where `astype` does (unless you pass `copy=False`, but then, just use `view`).

Comment (chungg) [inline gnocchi/carbonara.py]: i don't think this works as expected. from what i read, it's just tries to reevaluate based on bytes. it is faster though.

```
In [62]: dt.view()
Out[62]: numpy.datetime64('2017-01-01')

In [63]: dt.view('datetime64[ns]')
Out[63]: numpy.datetime64('1970-01-01T00:00:00.000017167')

In [64]: dt.astype('datetime64[ns]')
Out[64]: numpy.datetime64('2017-01-01T00:00:00.000000000')
```

Comment (jd) [inline gnocchi/carbonara.py]: Oh sh**. Ok. Maybe `copy=False`?
No big deal though.

Comment (chungg) [inline gnocchi/carbonara.py]: hmm. copy=False should work... but it gives me an error with numpy 1.12.1

```
In [85]: td.astype('timedelta64[ns]', copy=False)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-85-17b625fb7ebb> in <module>()
----> 1 td.astype('timedelta64[ns]', copy=False)

TypeError: astype() takes no keyword arguments
```

**Pull request merged**